### PR TITLE
Redesigning the create of event on mobile screen

### PR DESCRIPTION
**Pull request by tiagolimpe**
**Feature/Bug description:** New view to the creation of events on mobile screen.

**Solution:** 

selection_018
selection_019
selection_020
selection_021
selection_022

**TODO/FIXME:** n/a

### CODE REVIEW

File: frontend/app.js [hunk 1]
```diff
@@ -89,6 +89,15 @@
                     }
                 }
             })
+            .state("app.user.create_event", {
```
Comment (ruanres) [inline frontend/app.js]: add this state to the STATES constant and call it here

File: frontend/app.js [hunk 2]
```diff
@@ -476,6 +485,24 @@
         });
     });
 
+    /**
+     * Function to intercept the access of pages that should be displayed only on mobile screens.
+     * @param {service} $transitions - Service of transitions states
+     */
+    app.run(function mobileInterceptor($transitions) {
+        const permitted_routes = [
+            'app.user.create_event'
```
Comment (ruanres) [inline frontend/app.js]: use STATES constant

File: frontend/event/create_event.html [hunk 3]
```diff
@@ -0,0 +1,206 @@
+<div style="overflow: auto;">
+    <div class="create-event-mobile-top-bar">
+        <md-button class="md-icon-button" ng-click="createEventCtrl.previousStep()">
+            <md-icon class="back-to-button">keyboard_arrow_left</md-icon>
+        </md-button>
+    </div>
+</div>
+<form name="createEvent" style="margin-top: 3.2em;">
```
Comment (ruanres) [inline frontend/event/create_event.html]: move style to css

File: frontend/event/create_event.html [hunk 4]
```diff
@@ -0,0 +1,206 @@
+<div style="overflow: auto;">
```
Comment (ruanres) [inline frontend/event/create_event.html]: move all styles to css

File: frontend/app.js [hunk 5]
```diff
@@ -476,6 +485,24 @@
...
+        const permitted_routes = [
+            STATES.CREATE_EVENT
+        ];
+
+        $transitions.onSuccess({
+            to: (state) => {
+                return !Utils.isMobileScreen(475) && _.includes(permitted_routes, state.name);
+            }
+        }, (transition) => {
+            transition.router.stateService.transitionTo(STATES.HOME);
```
Comment (JuliePessoa) [inline frontend/app.js]: create a page that show "Esta página só está disponível para mobile" and a button to go to home state

File: frontend/event/create_event.html [hunk 6]
```diff
@@ -0,0 +1,204 @@
+<div class="event-overflow-navbar">
+    <div class="create-event-mobile-top-bar">
```
Comment (JuliePessoa) [inline frontend/event/create_event.html]: why a div with other div inside with just one component?
cant it just be only one div?

File: frontend/event/eventController.js [hunk 7]
```diff
@@ -62,19 +62,23 @@
...
-                targetEvent: event,
-                clickOutsideToClose: true,
-                locals: {
-                    events: eventCtrl.events
-                },
-                bindToController: true
-            }).then(() => {
-                eventCtrl._getEventsByDay();
-            });
+            if(!Utils.isMobileScreen(475)) {
```
Comment (JuliePessoa) [inline frontend/event/eventController.js]: create an app constant for this number

File: frontend/event/eventController.js [hunk 8]
```diff
@@ -62,19 +62,23 @@
...
+                $mdDialog.show({
```
Comment (JuliePessoa) [inline frontend/event/eventController.js]: i think could be better to read if it was `if mobile scren: return $state.go.``return $mddialog ....`

File: frontend/main/mainController.js [hunk 9]
```diff
@@ -14,7 +14,13 @@
         mainCtrl.pendingManagerMember = 0;
         mainCtrl.pendingInstInvitations = 0;
         mainCtrl.pendingInstLinksInvitations = 0;
-       
+
+        mainCtrl.stateView =  $state.current.name;
```
Comment (JuliePessoa) [inline frontend/main/mainController.js]: where is it being used?

File: frontend/event/eventDialogController.js [hunk 10]
```diff
@@ -4,7 +4,7 @@
     const app = angular.module("app");
 
     app.controller('EventDialogController', function EventDialogController(MessageService, brCidadesEstados,
```
Comment (JuliePessoa) [inline frontend/event/eventDialogController.js]: test the functions you modified/created

File: frontend/event/eventDialogController.js [hunk 11]
```diff
@@ -206,7 +213,17 @@
         };
 
         dialogCtrl.isValidStepOne = function isValidStepOne() {
-            return dialogCtrl.isValidAddress() && dialogCtrl.isValidDate();
+            const isMobileScreen = Utils.isMobileScreen(SCREEN_SIZES.SMARTPHONE);
```
Comment (JuliePessoa) [inline frontend/event/eventDialogController.js]: test

File: frontend/event/eventDialogController.js [hunk 12]
```diff
@@ -252,11 +274,12 @@
         }
 
         dialogCtrl.previousStep = function previousStep() {
-            var currentStep = _.findIndex(dialogCtrl.steps, function (situation) {
+            if (dialogCtrl.getStep(1)) dialogCtrl.cancelCreation();
```
Comment (JuliePessoa) [inline frontend/event/eventDialogController.js]: test

File: frontend/test/specs/event/eventDialogCtrlSpec.js [hunk 13]
```diff
@@ -315,17 +308,190 @@
...
+        expect(controller.getStreetPlaceholder()).toEqual(street + " *");
+      });
+    });
+
+    describe('_loadStateParams()', () => {
+      beforeEach(() => {
+        state.params.event = event;
+        state.params.events = [event];
+        state.params.isEditing = true;
+        controller.event = null;
```
Comment (JuliePessoa) [inline frontend/test/specs/event/eventDialogCtrlSpec.js]: i do prefer to use undefined.
and for verification you could use toBe(undefined)

File: frontend/event/eventDialogController.js [hunk 14]
```diff
@@ -268,6 +291,66 @@
...
+            dialogCtrl.event.end_time = new Date(dialogCtrl.event.start_time);
+            dialogCtrl.endHour = new Date(dialogCtrl.event.start_time);
+            dialogCtrl.endHour.setHours(18, 0, 0);
+            dialogCtrl.addEndHour();
+        };
+
+        /**
+         * Loads the current states of event to init the edition.
+         */
+        dialogCtrl._loadStatesToEdit = () => {
```
Comment (JuliePessoa) [inline frontend/event/eventDialogController.js]: test

File: frontend/test/specs/event/eventDialogCtrlSpec.js [hunk 15]
```diff
@@ -315,17 +308,190 @@
...
+        controller._loadEvent(event.key);
+        expect(eventService.getEvent).toHaveBeenCalledWith(event.key);
+        expect(controller.event).toEqual(event);
+      });
+
+      it('Should call state.go if the user is not the author of event', () => {
+        spyOn(state, 'go');
+        event.author_key = '000';
+        controller._loadEvent(event.key);
+        expect(state.go).toHaveBeenCalled();
```
Comment (JuliePessoa) [inline frontend/test/specs/event/eventDialogCtrlSpec.js]: verify parameter

File: frontend/test/specs/event/eventDialogCtrlSpec.js [hunk 16]
```diff
@@ -315,17 +308,190 @@
...
+        expect(controller.event.end_time.getHours()).not.toEqual(18);
+        controller.setToEntireDay();
+        expect(controller.event.start_time.getHours()).toEqual(8);
+        expect(controller.event.start_time.getMinutes()).toEqual(0);
+        expect(controller.event.end_time.getHours()).toEqual(18);
+        expect(controller.event.end_time.getMinutes()).toEqual(0);
+      });
+    });
+
+    describe('_loadEvent()', () => {
```
Comment (JuliePessoa) [inline frontend/test/specs/event/eventDialogCtrlSpec.js]: test error

File: frontend/test/specs/event/eventDialogCtrlSpec.js [hunk 17]
```diff
@@ -315,17 +308,190 @@
...
+      });
+
+      it('should call _loadStatesToEdit if have an event', () => {
+        spyOn(controller, '_loadStatesToEdit');
+        controller.event = event;
+        controller.$onInit();
+        expect(controller._loadStatesToEdit).toHaveBeenCalled();
+      });
+
+      it('should call _loadEvent if not have an event and have eventKey in state params', () => {
```
Comment (JuliePessoa) [inline frontend/test/specs/event/eventDialogCtrlSpec.js]: test the 'else' flow

File: frontend/event/eventDialogController.js [hunk 18]
```diff
@@ -252,11 +285,12 @@
         }
 
         dialogCtrl.previousStep = function previousStep() {
-            var currentStep = _.findIndex(dialogCtrl.steps, function (situation) {
+            if (dialogCtrl.getStep(1)) dialogCtrl.cancelCreation();
```
Comment (JuliePessoa) [inline frontend/event/eventDialogController.js]: shouldnt have a return on dialogCtrl.cancel Creation?

**Pull request merged**